### PR TITLE
Make `pandas` optional

### PR DESCRIPTION
**Pull request by arjxn-py**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Type of change

Please add a line in the relevant section of [CHANGELOG.md](https://github.com/pybamm-team/PyBaMM/blob/develop/CHANGELOG.md) to document the change (include PR #) - note reverse order of PR #s. If necessary, also add to the list of breaking changes.

- [ ] New feature (non-breaking change which adds functionality)
- [ ] Optimization (back-end change that speeds up the code)
- [ ] Bug fix (non-breaking change which fixes an issue)

# Key checklist:

- [ ] No style issues: `$ pre-commit run` (see [CONTRIBUTING.md](https://github.com/pybamm-team/PyBaMM/blob/develop/CONTRIBUTING.md#installing-and-using-pre-commit) for how to set this up to run automatically when committing locally, in just two lines of code)
- [ ] All tests pass: `$ python run-tests.py --all`
- [ ] The documentation builds: `$ python run-tests.py --doctest`

You can run unit and doctests together at once, using `$ python run-tests.py --quick`.

## Further checks:

- [ ] Code is commented, particularly in hard-to-understand areas
- [ ] Tests added that prove fix is effective or that feature works

### CODE REVIEW

**Review by Saransh-cpp — COMMENTED** [review]
Requesting a review from @brosaplanella and @tinosulzer here because they had a few concerns over this 🙂 

Refer - https://github.com/pybamm-team/PyBaMM/issues/3035#issuecomment-1591384803, https://github.com/pybamm-team/PyBaMM/issues/3035#issuecomment-1591990016

If we only require reading `csv` files using a non-pandas way to make `pandas` optional - you can look into the [`csv`](https://pypi.org/project/csv/) library which is bundled as a standard library in Python.

**Comment by Saransh-cpp** [comment]
@arjxn-py, let us know if you need help with this. You might need to explore the options listed above.

File: setup.py [hunk 1]
```diff
@@ -257,6 +256,7 @@ def compile_KLU():
         "all": [
             "anytree>=2.4.3",
             "autograd>=1.2",
+            "pandas>=0.24",
```
Comment (Saransh-cpp) [inline setup.py:259]: Should we also make a "pandas" group? It looks like users would be needing `pandas` more frequently, and installing all the dependencies to get `pandas` might not be nice.

Comment (arjxn-py) [reply setup.py:259]: Yes, i feel it's a good idea. I'd create a group for `pandas` too & update the docs accordingly.

**Review by tinosulzer — APPROVED** [review]
Looks good to me once pandas group is added

**Review by Saransh-cpp — APPROVED** [review]
@all-contributors please add @arjxn-py for code, docs

File: docs/source/user_guide/installation/index.rst [hunk 2]
```diff
@@ -95,6 +94,19 @@ Dependency                                                  Minimum Version    p
 `matplotlib <https://matplotlib.org/stable/>`__             2.0.0              plot               To plot various battery models, and analyzing battery performance.
 =========================================================== ================== ================== ================================================================== 
 
+.. _install.plot_dependencies:
+
+Pandas dependencies
+^^^^^^^^^^^^^^^^^
```
Comment (Saransh-cpp) [inline docs/source/user_guide/installation/index.rst:100]:
```suggestion
Pandas dependencies
^^^^^^^^^^^^^^^^^^^
```

File: docs/source/user_guide/installation/index.rst [hunk 3]
```diff
@@ -95,6 +94,19 @@ Dependency                                                  Minimum Version    p
 `matplotlib <https://matplotlib.org/stable/>`__             2.0.0              plot               To plot various battery models, and analyzing battery performance.
 =========================================================== ================== ================== ================================================================== 
 
+.. _install.plot_dependencies:
```
Comment (Saransh-cpp) [inline docs/source/user_guide/installation/index.rst:97]:
```suggestion
.. _install.pandas_dependencies:
```

**Comment by Saransh-cpp** [comment]
@all-contributors please add @arjxn-py for code and docs

**Pull request merged**